forem/forem
# [15 min fix] Fixed app code that was causing flaky E2E test

## What type of PR is this? (check all applicable)


## Description

Upon inspection, the E2E test is not flaky and in fact caught some brittle code we had in the front-end code base.

The code changes fix the issue and also give us a little performance boost. The way the code was previously, we registered n events for n comments on an article. Now there is only one event on the containing element of the comments section. With a few comments, not a big deal, but if there are say 50 comments, which I've definitely seen on posts, 1 event being registered instead of 50 is definitely a good thing.

## Related Tickets & Documents

Closes #13672 

## QA Instructions, Screenshots, Recordings

1. Navigate to a post created by you or from seed data on your local.
2. Comment on the post.
3. The new comment will be the top most comment in the Discussion section.
4. Click on the the three dot button at the top right of the newly created comment.
5. The menu should open.

### UI accessibility concerns?

N/A

## Added tests?

- [x] No, and this is why: The reenabled once thought was flaky test covers this.

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![Someone saying "Mystery solved!"](https://media.giphy.com/media/dtBUUa0dEvFgiL1OVx/giphy.gif)
